readthedocs/builds/views.py

```diff
@@ -148,6 +157,37 @@ class BuildDetail(BuildBase, DetailView):
 
     pk_url_kwarg = 'build_pk'
 
+    @method_decorator(login_required)
+    def post(self, request, project_slug, build_pk):
+        project = get_object_or_404(Project, slug=project_slug)
+        build = get_object_or_404(Build, pk=build_pk)
+
+        if not AdminPermission.is_admin(request.user, project):
+            return HttpResponseForbidden()
+
+        # NOTE: `terminate=True` is required for the child to attend our call
+        # immediately when it's running the build. Otherwise, it finishes the
+        # task. However, to revoke a task that has not started yet, we don't
+        # need it.
+        if build.state == BUILD_STATE_TRIGGERED:
+            # Since the task won't be executed at all, we need to update the
+            # Build object here.
+            terminate = False
+            build.state = BUILD_STATE_FINISHED
+            build.success = False
+            build.error = BuildCancelled.message
+            build.save()
+        else:
+            # In this case, we left the update of the Build object to the task
+            # itself to be executed in the `on_failure` handler.
+            terminate = True
+
+        app.control.revoke(build.task_id, signal=signal.SIGINT, terminate=terminate)
```

Comment:
This could be a helper `cancel_build` that we could use from other places as well.